File: src/entities/player.cs
using Godot;
using System;

public class player : KinematicBody2D {
	public float speed = 50F;
	public float runMod = 4F;

	private Vector2 velocity = Vector2.Zero;

	public override void _Ready() {
		GD.PrintS(this.Name, "reporting for duty");
	}

	public override void _PhysicsProcess(float delta) {
		movement(delta);
	}

	public void movement(float delta) {
		float speedMod = 1F;
		velocity = Vector2.Zero;

		if (Input.IsActionPressed("move.up")) velocity += Vector2.Up;
		if (Input.IsActionPressed("move.down")) velocity += Vector2.Down;
		if (Input.IsActionPressed("move.left")) velocity += Vector2.Left;
		if (Input.IsActionPressed("move.right")) velocity += Vector2.Right;

		if (Input.IsActionPressed("move.run")) speedMod = runMod;

		velocity = velocity.Normalized();

		MoveAndSlide(velocity * speed * speedMod);
	}
}

File: src/entities/player.nim
import godot
import godotapi / [kinematic_body_2d, input]

gdobj Player of KinematicBody2D:
  var speed = 50.0
  var runMod = 4.0

  var velocity: Vector2

  method ready*() =
    print(self.name, " ", "reporting for duty")

  method physicsProcess*(delta: float64) =
    self.movement(delta)

  method movement(delta: float64) {.base.} =
    var speedMod = 1.0
    self.velocity = vec2(0, 0)

    if input.isActionPressed("move.up"): self.velocity += vec2(0, -1)
    if input.isActionPressed("move.down"): self.velocity += vec2(0, 1)
    if input.isActionPressed("move.left"): self.velocity += vec2(-1, 0)
    if input.isActionPressed("move.right"): self.velocity += vec2(1, 0)

    if input.isActionPressed("move.run"): speedMod = self.runMod

    self.velocity = self.velocity.normalized()

    discard self.moveAndSlide(self.velocity * self.speed * speedMod)
